src/wp-includes/class-wp-user.php

```diff
@@ -203,6 +203,10 @@ public static function get_data_by( $field, $value ) {
 			$field = 'id';
 		}
 
+		if ( ! $value ) {
+			return false;
+		}
+
```

Comment:
Adding this is fine and required. 

Removing the lower one has the potential of breaking things.

Example: `! ' 0'` is `false`, while `! trim( ' 0' )` is `true`

So removing lines 219-222 could potentially change the  result of `get_data_by()`